include/forcing/NullForcingProvider.hpp

```diff
@@ -0,0 +1,271 @@
+#ifndef NGEN_NULLFORCING_H
+#define NGEN_NULLFORCING_H
+
+#include <vector>
+#include <set>
+#include <cmath>
+#include <algorithm>
+#include <string>
+#include <fstream>
+#include <iostream>
+#include <unordered_map>
+#include <boost/algorithm/string.hpp>
+#include <boost/lexical_cast.hpp>
+#include "CSV_Reader.h"
+#include <ctime>
+#include <time.h>
+#include <memory>
+#include "AorcForcing.hpp"
+#include "GenericDataProvider.hpp"
+#include "DataProviderSelectors.hpp"
+#include <exception>
+#include <UnitsHelper.hpp>
+
+/**
+ * @brief Forcing class providing time-series precipiation forcing data to the model.
+ */
+class NullForcingProvider : public data_access::GenericDataProvider
+{
+    public:
+
+    typedef struct tm time_type;
+
+
+    NullForcingProvider(forcing_params forcing_config):start_date_time_epoch(forcing_config.simulation_start_t),
+                                           end_date_time_epoch(forcing_config.simulation_end_t),
+                                           current_date_time_epoch(forcing_config.simulation_start_t),
+                                           forcing_vector_index(-1)
+    {
+        read_csv();
+    }
+
+    // BEGIN DataProvider interface methods
+
+    /**
+     * @brief the inclusive beginning of the period of time over which this instance can provide data for this forcing.
+     *
+     * @return The inclusive beginning of the period of time over which this instance can provide this data.
+     */
+    long get_data_start_time() override {
+        //FIXME: Trace this back and you will find that it is the simulation start time, not having anything to do with the forcing at all.
+        // Apparently this "worked", but at a minimum the description above is false.
+        //return start_date_time_epoch;
+        // LONG_MIN is a large negative number
+        return 0;
+    }
+
+    /**
+     * @brief the exclusive ending of the period of time over which this instance can provide data for this forcing.
+     *
+     * @return The exclusive ending of the period of time over which this instance can provide this data.
+     */
+    long get_data_stop_time() override {
+        //return end_date_time_epoch;
+        return LONG_MAX;
+    }
+
+    /**
+     * @brief the duration of one record of this forcing source
+     *
+     * @return The duration of one record of this forcing source
+     */
+    long record_duration() override {
+        //return time_epoch_vector[1] - time_epoch_vector[0];
+        //TODO find a more general way to set it
+        long timestep_size = 3600;
```

Comment:
This should probably be `1` or even `0`? It may not matter, but this provider doesn't really _have_ a timestep.

Reply:
I'll probably use ```1``` but will test ```0``` to see what it does.